### PR TITLE
Show more data in Network exceptions

### PR DESCRIPTION
**Pull request by EvgenyAgafonchikov**
## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Network/Commands.Network/Common/NetworkCloudException.cs [hunk 1]
```diff
@@ -51,39 +51,53 @@ protected static string GetErrorMessageWithRequestIdInfo(Rest.Azure.CloudExcepti
...
                     return sb.ToString();
                 }
-                
+
+                if (errorReturned.Error.Code != null)
+                {
+                    sb.AppendLine().AppendFormat("ErrorCode: {0}", errorReturned.Error.Code);
+                }
+
+                if (errorReturned.Error.Code != null)
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/Common/NetworkCloudException.cs]: This check should be for `errorReturned.Error.Target`

Also, what information does the `Error.Target` convey?

**Comment by praries880** [comment]
@EvgenyAgafonchikov Is it possible to add tests for the changes made here?

File: src/ResourceManager/Network/Commands.Network/Common/NetworkCloudException.cs (hunk 1 above)
Comment (EvgenyAgafonchikov) [reply src/ResourceManager/Network/Commands.Network/Common/NetworkCloudException.cs]: This seems to be similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SdkCommon/ClientRuntime.Azure/ClientRuntime.Azure/CommonModels/CloudError.cs#L37) but supposed to be deserialized from JSON response, though there is not much description in SDK.
I also didn't manage to trigger error that adds value there to get more details. In general, we would never show it if it is not present.

**Comment by praries880** [comment]
@EvgenyAgafonchikov 
Can you add a change log entry for the change under [here](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Network/Commands.Network/ChangeLog.md#current-release)?
Otherwise the changes look good to me.

**Comment by EvgenyAgafonchikov** [comment]
@praries880, changelog was updated, please have another look

**Pull request merged**